src/engine/execute.js

```diff
@@ -85,8 +87,17 @@ const execute = function (sequencer, thread) {
         } else {
             // In a non-hat, report the value visually if necessary if
             // at the top of the thread stack.
-            if (typeof resolvedValue !== 'undefined' && thread.showVisualReport && thread.atStackTop()) {
-                runtime.visualReport(currentBlockId, resolvedValue);
+            if (typeof resolvedValue !== 'undefined' && thread.atStackTop()) {
+                if (thread.showVisualReport) {
+                    runtime.visualReport(currentBlockId, resolvedValue);
+                }
+
+                if (thread.updateMonitor) {
+                    runtime.updateMonitors([{
+                        id: currentBlockId,
+                        value: String(resolvedValue)
+                    }]);
```

Comment:
One way to aggregate these changes would be to have this method just update the monitor state, rather than firing the event.